Lab3/ComputerStore.WebUI/ApiControllers/BuyersApiController.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Threading.Tasks;
+using ComputerStore.BusinessLogicLayer.Managers;
+using ComputerStore.BusinessLogicLayer.Models;
+using ComputerStore.DataAccessLayer.Models.Identity;
+using ComputerStore.WebUI.AppConfiguration;
+using ComputerStore.WebUI.Controllers;
+using ComputerStore.WebUI.Mappers;
+using ComputerStore.WebUI.Models;
+using ComputerStore.WebUI.Models.JwtToken;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace ComputerStore.WebUI.ApiControllers
+{
+    [Route("api/buyers")]
+    [ApiController]
+    [Authorize(Roles = RolesNames.AdminOrUser, AuthenticationSchemes = JwtInfo.AuthSchemes)]
+    public class BuyersApiController : ControllerBase
+    {
+        private readonly BuyerManager _buyerManager;
+        private readonly ILogger<BuyersApiController> _logger;
+        private readonly UserManager<IdentityBuyer> _userManager;
+
+        public BuyersApiController(BuyerManager buyerManager,
+            ILogger<BuyersApiController> logger,
+            UserManager<IdentityBuyer> userManager)
+        {
+            _buyerManager = buyerManager;
+            _logger = logger;
+            _userManager = userManager;
+        }
+
+        [HttpGet("details")]
+        public async Task<BuyerViewModel> Details()
```

Comment:
It's bad practice to return ViewModels here because it's related to the UI only.
Return BLL models.